### PR TITLE
Added all new credential related routes

### PR DESCRIPTION
**Pull request by jenow**
Creation of new actions in auth controller:
- [x] deleteMyCredentials
- [x] createMyCredentials
- [x] getMyCredentials
- [x] hasMyCredentials
- [x] updateMyCredentials
- [x] validateMyCredentials
Creation of new actions in security controller:
- [x] createCredentials
- [x] deleteCredentials
- [x] getCredentialFields
- [x] getCredentials
- [x] getAllCredentialFields
- [x] hasCredentials
- [x] updateCredentials
- [x] validateCredentials

### CODE REVIEW

File: src/security/Security.js [hunk 1]
```diff
@@ -751,4 +751,197 @@ Security.prototype.getUserRights = function (userId, options, cb) {
...
+ * Create credentials of the specified <strategy> for the user <kuid>.
+ *
+ * @param strategy
+ * @param kuid
+ * @param credentials
+ * @param options
+ * @param cb
+ * @returns {Security}
+ */
+Security.prototype.createCredentials = function (strategy, kuid, credentials, options, cb) {
```
Comment (xbill82) [inline src/security/Security.js]: What if `!strategy`?

File: src/security/Security.js [hunk 2]
```diff
@@ -751,4 +751,197 @@ Security.prototype.getUserRights = function (userId, options, cb) {
...
+/**
+ * Delete credentials of the specified <strategy> for the user <kuid> .
+ *
+ * @param strategy
+ * @param kuid
+ * @param options
+ * @param cb
+ * @returns {Security}
+ */
+Security.prototype.deleteCredentials = function (strategy, kuid, options, cb) {
```
Comment (xbill82) [inline src/security/Security.js]: What if `!strategy`?

Comment (xbill82) [inline src/security/Security.js]: This applies to all the following methods that accept `strategy` as the first argument.

Comment (jenow) [inline src/security/Security.js]: Then kuzzle wil answer an error

Comment (xbill82) [inline src/security/Security.js]: ok then we assume that we leave the responsibility to Kuzzle. Approving.

**Pull request merged**